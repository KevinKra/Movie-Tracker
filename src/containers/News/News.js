import React, { Component } from "react";
import { ReactComponent as Arrow } from "../../assets/arrow.svg";
import { fetchNews } from "../../actions";
import { connect } from "react-redux";
import "./News.scss";

export class News extends Component {
  state = { slidesPosition: 5 };

  changeSlidesPositionForward = () => {
    const { slidesPosition } = this.state;
    let position = slidesPosition;
    if (position < 30) {
      position++;
      this.setState({ slidesPosition: position });
    } else {
      this.setState({ slidesPosition: 0 });
    }
  };

  changeSlidesPositionBack = () => {
    const { slidesPosition } = this.state;
    let position = slidesPosition;
    if (slidesPosition === 0) {
      this.setState({ slidesPosition: 30 });
    } else {
      position--;
      this.setState({ slidesPosition: position });
    }
  };

  render = () => {
    const slideDisplay = this.props.news.map((stories, index) => {
      const result =
        stories.multimedia.length > 1
          ? Object.values(stories.multimedia[4])
          : null;
      const newArr = result ? result[0] : null;

      let backgroundImage = {
        backgroundImage: `url(${newArr})`
      };
      if (index === this.state.slidesPosition) {
        return (
          <div className="slide-image" style={backgroundImage} id={stories.id}>
            <h4>{stories.title}</h4>
          </div>
        );
      } else {
        return null;
      }
    });
    return (
      <div className="slideshow-section">
        <div className="slide-container">
          {slideDisplay}
          <Arrow
            className="arrow-one"
            onClick={this.changeSlidesPositionForward}
          />
          <Arrow
            className="arrow-two"
            onClick={this.changeSlidesPositionBack}
          />
        </div>
      </div>
    );
  };
}

const mapStateToProps = state => {
  return {
    news: state.news
  };
};

export default connect(
  mapStateToProps,
  { fetchNews }
)(News);
