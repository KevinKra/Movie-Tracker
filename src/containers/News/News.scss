@import "../../variables.scss";

.slideshow-section {
  height: 65%;
  div {
    position: relative;
    height: 525px;
    width: 520px;
  }

  .slide-image {
    filter: brightness(0.75);
    box-sizing: border-box;
    position: relative;
    background-position: top center;
    background-size: cover;
    width: 100%;
    height: 100%;
    display: table;
    padding: 40px;
    h4 {
      position: relative;
      z-index: 3;
      font-size: 1.8rem;
      text-shadow: 2px 2px 5px black;
      display: table-cell;
      vertical-align: bottom;
      color: white;
    }
  }
  .slide-container {
    position: relative;
    border-radius: 5px;
    .slide {
      text-decoration: none;
    }
    &::after {
      content: "";
      top: 0px;
      right: 0px;
      position: absolute;
      width: 100px;
      z-index: 2;
      height: 100%;
    }
    &::before {
      content: "";
      left: 0px;
      z-index: 2;
      height: 100%;
    }
  }

  .arrow {
    z-index: 3;
    fill: #c7c7c7;
    height: 50px;
    width: 70px;
    top: 45%;
    right: 0px;
    filter: drop-shadow(1px 1px 3px $background-color);
    position: absolute;
    &:hover {
      cursor: pointer;
      fill: rgb(226, 226, 226);
    }
  }

  .arrow-one {
    transform: rotate(180deg) scale(0.8);
    transition: 0.5s;
    &:hover {
      transform: rotate(180deg) scale(0.85);
    }
  }
  .arrow-two {
    top: 45%;
    left: 0px;
    transform: scale(0.8);
    transition: 0.5s;
    &:hover {
      transform: scale(0.85);
    }
  }
}
