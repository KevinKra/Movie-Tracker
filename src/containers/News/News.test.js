import React from "react";
import { shallow } from "enzyme";
import { News } from "./News";
import MockData from "../../assets/mockData";

let mockNews = MockData.newsResults;

describe("News", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<News news={mockNews} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
