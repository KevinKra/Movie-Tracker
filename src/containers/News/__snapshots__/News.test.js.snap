// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News should match the snapshot 1`] = `ShallowWrapper {}`;
